### PR TITLE
Update crash reporting settings strings

### PR DESCRIPTION
**Pull request by atuchin-m**
Resolves https://github.com/brave/brave-browser/issues/19822

We are planning to enable `HangWatcher` feature which send crash dump when browser hangs.
Therefore we need to update string in settings.

The result:
image

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: script/lib/grd_string_replacements.py [hunk 1]
```diff
@@ -22,9 +22,9 @@
 default_replacements = [
     (r'Brave Web Store', r'Web Store'),
     (r'Automatically send usage statistics and crash reports to Brave',
-     r'Automatically send crash reports to Brave'),
+     r'Automatically send reports to Brave when the browser crashes or hangs'),
     (r'Automatically sends usage statistics and crash reports to Brave',
-     r'Automatically sends crash reports to Brave'),
+     r'Automatically sends reports to Brave when the browser crashes or hangs'),
```
Comment (atuchin-m) [inline script/lib/grd_string_replacements.py:27]: In fact the diff in this file is only one primary change in the PR.
All the rest changes are generated by `npm run chromium_rebase_l10n`.
@Max @mariospr Is it the expected way to update such strings?

Comment (mkarolin) [reply script/lib/grd_string_replacements.py:27]: I think this is fine as the strings are already in here and you are just tweaking them more.

**Review by iefremov — APPROVED** [review]

**Comment by atuchin-m** [comment]
The texts have been updated according to [Pat's comment](https://github.com/brave/brave-com-resources/issues/424#issuecomment-994702913).

**Comment by atuchin-m** [comment]
Here are the new screenshots
new_desktop_crash_settings
new_android_settings
new_after_crash_dialog
.

File: app/generated_resources.grd [hunk 2]
```diff
@@ -7474,7 +7474,7 @@ Keep your key file in a safe place. You will need it to create new versions of y
 
       <!-- First-run dialog -->
       <message name="IDS_FR_ENABLE_LOGGING" desc="The label of the checkbox to enable/disable crash and user metrics logging">
-        Automatically send crash reports to Brave
+        Automatically send usage statistics and crash reports to Brave
```
Comment (atuchin-m) [inline app/generated_resources.grd:7477]: This string isn't used in Brave. Now it's almost reverted to the original value (expect autoreplace `Chrome => Brave`)

File: browser/ui/android/strings/android_brave_strings.grd [hunk 3]
```diff
@@ -315,11 +315,11 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
       <message name="IDS_SEND_P3A_ANALYTICS_SUMMARY" desc="Summary for checkbox of send p3a analytics.">
         Anonymized P3A info helps Brave estimate overall usage, and ensure we’re improving popular features.
       </message>
-      <message name="IDS_SEND_CRASH_REPORTS_TITLE" desc="Title for checkbox of send crash reports automatically.">
-        Automatically send crash reports
+      <message name="IDS_SEND_CRASH_REPORTS_TITLE" desc="Title for for checkbox enable/disable sending diagnostic reports (crashes, hangs).">
```
Comment (goodov) [inline browser/ui/android/strings/android_brave_strings.grd:318]: typo: for for

**Review by goodov — APPROVED** [review]
lgtm, but fix a typo pls.

File: browser/ui/android/strings/android_brave_strings.grd (hunk 3 above)
Comment (atuchin-m) [reply browser/ui/android/strings/android_brave_strings.grd:318]:
```suggestion
      <message name="IDS_SEND_CRASH_REPORTS_TITLE" desc="Title for checkbox enable/disable sending diagnostic reports (crashes, hangs).">
```

**Pull request merged**